gcp/modules/gpii-istio/network_policy.tf

```diff
@@ -5,9 +5,57 @@ resource "kubernetes_network_policy" "deny-default" {
   }
 
   spec {
-    pod_selector = {}
+    pod_selector {}
+
+    policy_types = ["Ingress", "Egress"]
+
+    egress {
+      # Allow egress traffic to Istio control plane
+      to {
+        namespace_selector {
+          match_labels = {
+            k8s-app = "istio"
+          }
+        }
+      }
+
+      # Allow egress traffic within namespace
+      to {
+        namespace_selector {
+          match_labels = {
+            name = "gpii"
+          }
+        }
+      }
+
+      # Allow egress traffic to dns
+      to {
+        namespace_selector {
+          match_labels = {
+            name = "kube-system"
+          }
+        }
+        pod_selector {
+          match_labels = {
+            k8s-app = "kube-dns"
+          }
+        }
+      }
+    }
+
+    egress {
+      # Allow egress traffic Compute Metadata
+      to {
+        ip_block {
+          cidr = "10.16.0.0/20"
```

Comment:
There are more networks in [cluster_subnets](https://github.com/gpii-ops/gpii-infra/blob/master/gcp/modules/gke-network/main.tf#L22), do they need to be included here as well?

Reply:
Nope, only the IPs of the nodes, because that's what the metadata traffic is being natted to (the other IPs are for services and pods).

Reply:
Ok, makes sense. Had to look into exekube's `gke-network` to understand what these extra networks are.